Source/WebCore/rendering/mac/RenderThemeMac.mm

```diff
@@ -1021,26 +1021,36 @@ static NSControlSize stepperControlSizeForFont(const FontCascade& font)
     return sizes;
 }
 
+static constexpr std::array switchMarginsNonMini { 2, 2, 1, 2 };
+static constexpr std::array switchMarginsMini { 1, 1, 0, 1 };
+
 static std::span<const int, 4> switchMargins(NSControlSize controlSize)
 {
     static constexpr std::array margins {
         // top right bottom left
-        std::array { 2, 2, 1, 2 },
-        std::array { 2, 2, 1, 2 },
-        std::array { 1, 1, 0, 1 },
-        std::array { 2, 2, 1, 2 },
+        switchMarginsNonMini,
+        switchMarginsNonMini,
+        switchMarginsMini,
+        switchMarginsNonMini
+    };
+    return margins[controlSize];
+}
+
+static std::span<const int, 4> switchMarginsVertical(NSControlSize controlSize)
```

Comment:
I'm confused why we just don't transpose once rather than duplicate the ordering 4 times?

Reply:
(once below in `visualSwitchMargins` like the existing code)